memguard_test.go

```diff
@@ -521,3 +505,43 @@ func TestGetBytes(t *testing.T) {
 		t.Error("pointer does not describe actual memory")
 	}
 }
+
+func TestFinalizer(t *testing.T) {
+	b, err := New(8, false)
+	if err != nil {
+		t.Error("unexpected error")
+	}
+	ib := b.lockedBuffer
+
+	c, err := New(8, false)
+	if err != nil {
+		t.Error("unexpected error")
+	}
+	ic := c.lockedBuffer
+
+	runtime.GC()
+	for ib.IsDestroyed() != true {
+		runtime.Gosched()
+	}
+
+	if ib.IsDestroyed() != true {
```

Comment:
This will never happen. For this to happen, the previous loop would be infinite. Need to rethink these tests.